.buildkite/release-builds.yml

```diff
@@ -10,7 +10,7 @@ common_params:
         repo: "woocommerce/woocommerce-ios/"
   # Common environment values to use with the `env` key.
   - &common_env
-    IMAGE_ID: xcode-13.4.1
+    IMAGE_ID: xcode-14
```

Comment:
This was updated for the `pipeline.yml` but not for the Release pipeline, which caused the initial 10.6 build to fail in CI.